From: Hannes Reinecke <hare@suse.de>
Date: Tue, 11 Apr 2017 14:22:22 +0200
Subject: megaraid_sas: Fixup tgtid count in megasas_ld_list_query()
References: bsc#971933
Patch-Mainline: Never, fix to a SUSE-specific patch

We never actually pull in the target ID count, leading to wrong
calculations when trying to enumerate devices.


Reported-by: Sumit Saxena <sumit.saxena@broadcom.com>
Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 drivers/scsi/megaraid/megaraid_sas_base.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/scsi/megaraid/megaraid_sas_base.c b/drivers/scsi/megaraid/megaraid_sas_base.c
index b00e304..0597d5f 100644
--- a/drivers/scsi/megaraid/megaraid_sas_base.c
+++ b/drivers/scsi/megaraid/megaraid_sas_base.c
@@ -4054,6 +4054,7 @@ megasas_ld_list_query(struct megasas_instance *instance, u8 query_type)
 	else
 		ret = megasas_issue_polled(instance, cmd);
 
+	tgtid_count = le32_to_cpu(ci->count);
 	if (tgtid_count == 0) {
 		/* No drives found, try the older LD list DCMD */
 		ret = 1;
-- 
1.8.5.6

